modules/apps/commerce/commerce-channel-web/test/js/commerce_marketplace_payment_method/index.spec.tsx

```diff
@@ -0,0 +1,428 @@
+
+		const addButton = queryByText('add');
+
+		await fireEvent.click(addButton as HTMLButtonElement);
+
+		await act(async () => jest.runAllTimers());
+
+		expect(queryByText('Map for Objects')).toBeTruthy();
+
+		await fireEvent.click(
+			queryByText('Map for Objects') as HTMLButtonElement
+		);
+
+		expect(queryByText('back-to-list')).toBeTruthy();
+
+		await act(
+			async () =>
+				await fireEvent.click(
+					queryByText('install') as HTMLButtonElement
+				)
+		);
+
+		expect(queryByText('confirmation-required')).toBeTruthy();
+
+		await act(
+			async () =>
+				await fireEvent.click(
+					queryByText('install') as HTMLButtonElement
+				)
+		);
+
+		expect(fetch.mock.calls.length).toBe(4);
+
+		expect(queryByText('there-was-an-unknown-error')).toBeTruthy();
+	});
+
+	it('renders authorized with apps and click install success', async () => {
+		const mockCartResponse = {
+			cartItems: [
+				{
+					id: 123,
+				},
+			],
+			id: 123,
+		};
+
+		fetch
+			.mockResponseOnce(JSON.stringify(marketplaceSettings))
+			.mockResponseOnce(JSON.stringify(productRequestRsponse))
+			.mockResponseOnce(JSON.stringify(mockCartResponse))
+			.mockResponseOnce(JSON.stringify(mockCartResponse))
+			.mockResponseOnce(JSON.stringify({ok: true}));
+
+		const {queryByText} = render(<CommerceChannelAddPaymentMethod />);
+
+		const addButton = queryByText('add');
+
+		await fireEvent.click(addButton as HTMLButtonElement);
+
+		await act(async () => jest.runAllTimers());
+
+		expect(queryByText('Map for Objects')).toBeTruthy();
+
+		await fireEvent.click(
+			queryByText('Map for Objects') as HTMLButtonElement
+		);
+
+		expect(queryByText('back-to-list')).toBeTruthy();
+
+		await act(
+			async () =>
+				await fireEvent.click(
+					queryByText('install') as HTMLButtonElement
+				)
+		);
+
+		expect(queryByText('confirmation-required')).toBeTruthy();
+
+		await act(
+			async () =>
+				await fireEvent.click(
+					queryByText('install') as HTMLButtonElement
+				)
+		);
+
+		expect(fetch.mock.calls.length).toBe(5);
+
+		expect(queryByText('success')).toBeTruthy();
+
+		expect(
+			queryByText(
+				'your-application-has-been-installed,-wait-a-few-moments-for-it-to-become-available'
+			)
+		).toBeTruthy();
+	});
+
+	it('renders authorized with apps and click install no resource', async () => {
+		const maketpalceSettingsTest = {
+			...marketplaceSettings,
+			data: {
+				...marketplaceSettings.data,
+				settings: {
+					...marketplaceSettings.data.settings,
+					cloudProject: 'NO_RESOURCE',
+				},
+			},
+		};
+
+		const mockCartResponse = {
+			cartItems: [
+				{
+					id: 123,
+				},
+			],
+			id: 123,
+		};
+
+		fetch
+			.mockResponseOnce(JSON.stringify(maketpalceSettingsTest))
+			.mockResponseOnce(JSON.stringify(productRequestRsponse))
+			.mockResponseOnce(JSON.stringify(mockCartResponse))
+			.mockResponseOnce(JSON.stringify(mockCartResponse))
+			.mockResponseOnce(JSON.stringify({ok: false}));
```

Comment:
Inline this variables

```suggestion
		const maketpalceSettingsTest = {
			...marketplaceSettings,
			data: {
				...marketplaceSettings.data,
				settings: {
					...marketplaceSettings.data.settings,
					cloudProject: 'NO_RESOURCE',
				},
			},
		};

		const mockCartResponse = {
			cartItems: [
				{
					id: 123,
				},
			],
			id: 123,
		};

		fetch
			.mockResponseOnce(JSON.stringify(maketpalceSettingsTest))
			.mockResponseOnce(JSON.stringify(productRequestRsponse))
			.mockResponseOnce(JSON.stringify(mockCartResponse))
			.mockResponseOnce(JSON.stringify(mockCartResponse))
			.mockResponseOnce(JSON.stringify({success: true}));
```